mssql-plugin/src/main/java/io/cdap/plugin/mssql/SqlServerSink.java

```diff
@@ -126,19 +144,26 @@ protected FieldsValidator getFieldsValidator() {
     @Nullable
     public String currentLanguage;
 
-    @Override
-    public String getConnectionString() {
-      return String.format(SqlServerConstants.SQL_SERVER_CONNECTION_STRING_FORMAT, host, port, database);
-    }
-
     @Override
     public Map<String, String> getDBSpecificArguments() {
-      return SqlServerUtil.composeDbSpecificArgumentsMap(instanceName, authenticationType, null,
+      return SqlServerUtil.composeDbSpecificArgumentsMap(instanceName, connection.getAuthenticationType(), null,
                                                          connectTimeout, columnEncryption, encrypt,
                                                          trustServerCertificate, workstationId, failoverPartner,
                                                          packetSize, queryTimeout);
     }
 
+    @Override
+    public String getConnectionString() {
+      return String
+        .format(SqlServerConstants.SQL_SERVER_CONNECTION_STRING_FORMAT, connection.getHost(), connection.getPort(),
```

Comment:
nit - we normally split on the start of parameters instead of on `.`

Reply:
This is not addressed

Reply:
Acknowledged.